### PR TITLE
Gui editor

### PR DESCRIPTION
**Pull request by NopoTheGamer**
dungeon map still uses /neumap

I would add a command but minecraft decided it doesnt like that idea, if someone else wants to work it out that would be great

I get a bug with the lapis thingy on the hollows overlay but it doesnt happen to jani so idk whats up with that

### CODE REVIEW

**Review by IRONM00N — APPROVED** [review]

**Review by hannibal002 — APPROVED** [review]

File: src/main/java/io/github/moulberry/notenoughupdates/core/config/Position.java [hunk 1]
```diff
@@ -31,6 +31,8 @@ public class Position {
 	private boolean centerX;
 	@Expose
 	private boolean centerY;
+	@Expose
```
Comment (romangraef) [inline src/main/java/io/github/moulberry/notenoughupdates/core/config/Position.java:34]: why expose this

Comment (NopoTheGamer) [reply src/main/java/io/github/moulberry/notenoughupdates/core/config/Position.java]: no clue

File: src/main/java/io/github/moulberry/notenoughupdates/core/config/gui/GuiPositionEditor.java [hunk 2]
```diff
@@ -29,45 +34,72 @@
...
+			pos.add(overlayPositions.get(overlay));
+			ogPos.add(pos.get(i).clone());
+			width.add((int) overlay.getDummySize().x);
+			height.add((int) overlay.getDummySize().y);
+			if (i < 10) {
+				renderCallbac.add(() -> {
+					overlay.renderDummy();
+					OverlayManager.dontRenderOverlay = overlay.getClass();
+				});
+			}
```
Comment (romangraef) [inline src/main/java/io/github/moulberry/notenoughupdates/core/config/gui/GuiPositionEditor.java:78]: what is this all about

**Comment by hannibal002** [comment]
There are some new features in master with overlays that need to changed to this new format.

File: src/main/java/io/github/moulberry/notenoughupdates/core/config/gui/GuiPositionEditor.java (hunk 2 above)
Comment (NopoTheGamer) [reply src/main/java/io/github/moulberry/notenoughupdates/core/config/gui/GuiPositionEditor.java]: Not sure what part you mean but I changed the if statement 
the render callback is what renders the text ( :scream: )

**Review by romangraef — DISMISSED** [review]

File: src/main/java/io/github/moulberry/notenoughupdates/core/config/gui/GuiPositionEditorButForTheDungeonMap.java [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.Gui;
+import net.minecraft.client.gui.GuiScreen;
+import net.minecraft.client.gui.ScaledResolution;
+import org.lwjgl.input.Keyboard;
+import org.lwjgl.input.Mouse;
+
+import java.io.IOException;
+
+public class GuiPositionEditorButForTheDungeonMap extends GuiScreen { // nea is gonna LOVE this file's existence
```
Comment (walkerselby) [inline src/main/java/io/github/moulberry/notenoughupdates/core/config/gui/GuiPositionEditorButForTheDungeonMap.java:33]: teehee

**Review by walkerselby — DISMISSED** [review]

**Review by hannibal002 — DISMISSED** [review]

**Review by hannibal002 — DISMISSED** [review]

**Review by romangraef — APPROVED** [review]

**Comment by NopoTheGamer** [comment]
tbh im not sure how the map fixes ended up in on this branch but :woman_shrugging:

**Review by hannibal002 — APPROVED** [review]

**Review by romangraef — APPROVED** [review]

**Review by Lulonaut — APPROVED** [review]

**Pull request merged**